pytket/pytket/circuit/logic_exp.py

```diff
@@ -350,22 +350,26 @@ class RegRsh(BinaryOp, RegLogicExp):
     op = RegWiseOp.RSH
 
 
-class ConstPredicate(BinaryOp):
-    """A binary predicate where at least one of the arguments is constant."""
+class RegPredicate(BinaryOp):
+    """A binary predicate where the arguments are either registers or constants."""
 
-    def __init__(self, exp: Union[LogicExp, Variable], const: Constant) -> None:
-        super().__init__(exp, const)
-        assert isinstance(exp, (LogicExp, Bit, BitRegister))
-        assert isinstance(const, Constant)
+    def __init__(
+        self,
+        lhs: Union[LogicExp, Variable, Constant],
+        rhs: Union[LogicExp, Variable, Constant],
+    ) -> None:
+        super().__init__(lhs, rhs)
+        assert isinstance(lhs, (LogicExp, Bit, BitRegister, Constant))
+        assert isinstance(rhs, (LogicExp, Bit, BitRegister, Constant))
```

Comment:
these assertions are now unecessary and can be removed

Reply:
In fact the whole init function can therefore be removed

Reply:
OK, I just pass it ?

Reply:
the whole function definition can be removed, it will call the super init automatically

Reply:
Yes, sure but what I meant is that:
```
class Predicate(BinaryOp):
"""A binary predicate where the arguments are either Bits, BitRegisters, or constants."""
 ...
```

Reply:
Oh really ? OK didn't know about that. Thanks.

Reply:
Done.